src/test/java/cohen/wordle/WordleControllerTest.java

```diff
@@ -0,0 +1,187 @@
+package cohen.wordle;
+
+import org.junit.jupiter.api.Test;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+
+import static cohen.wordle.CharStatus.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class WordleControllerTest
+{
+    WordleGame wordleGame = mock();
+    JLabel[][] letters = new JLabel[][]
+            {
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()}
+            };
+
+    StringBuilder guess = mock();
+
+    @Test
+    public void addLetter()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.addLetter("t");
+        guess.append("t");
+        controller.addLetter("a");
+        guess.append("a");
+        controller.addLetter("u");
+        guess.append("u");
+
+        //then
+        verify(letters[0][0]).setText("T");
+        verify(guess).append("t");
+        verify(letters[0][1]).setText("A");
+        verify(guess).append("a");
+        verify(letters[0][2]).setText("U");
+        verify(guess).append("u");
+    }
+
+    @Test
+    public void backspaceAfterNone()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.backspace();
+
+        //then
+        assertNull(letters[0][0].getText());
```

Comment:
This is going to be null because this label is a mock. You aren't actually testing anything here.